### PR TITLE
Add training status support

### LINKED ISSUE #129: add training status
daily: `/mobile-gateway/usersummary/trainingstatus/latest/2025-06-11`
weekly: `/mobile-gateway/usersummary/trainingstatus/weekly/2025-05-15/2025-06-11`
monthly: `/mobile-gateway/usersummary/trainingstatus/monthly/2024-12-12/2025-06-11`

### PR DESCRIPTION
**Pull request by matin**
## Summary
- Add DailyTrainingStatus, WeeklyTrainingStatus, MonthlyTrainingStatus classes
- Custom parsing logic for complex nested API responses  
- Comprehensive test coverage with VCR cassettes
- Documentation in README with usage examples

## API Endpoints Added
- Daily: `/mobile-gateway/usersummary/trainingstatus/latest/{date}`
- Weekly: `/mobile-gateway/usersummary/trainingstatus/weekly/{start}/{end}`
- Monthly: `/mobile-gateway/usersummary/trainingstatus/monthly/{start}/{end}`

## Implementation Details
- Follows existing patterns in the stats module for consistency
- Takes advantage of current base classes to keep code DRY
- Custom parsing logic handles the complex nested response structure
- All fields are properly typed with Pydantic dataclasses

## Test Coverage
- 5 comprehensive tests covering daily, weekly, monthly endpoints
- Tests include pagination scenarios and edge cases
- VCR cassettes recorded for reliable test execution

Closes #129

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/garth/stats/training_status.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                        acute_load = entry.pop("acuteTrainingLoadDTO", {})
+                        if isinstance(acute_load, dict):
+                            entry.update(acute_load)
+                        result.append(entry)
+                return result
+
+        return []
+
+    @classmethod
+    def list(
```
Comment (matin) [inline src/garth/stats/training_status.py:97]: this appears to replicate much of the functionality is the base class for stats

File: src/garth/stats/training_status.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    _path: ClassVar[str] = f"{BASE_PATH}/weekly/{{start}}/{{end}}"
+    _data_key: ClassVar[str] = "weeklyTrainingStatus"
+    _page_size: ClassVar[int] = 52
+
+
+@dataclass
+class MonthlyTrainingStatus(TrainingStatus):
+    _path: ClassVar[str] = f"{BASE_PATH}/monthly/{{start}}/{{end}}"
+    _data_key: ClassVar[str] = "monthlyTrainingStatus"
+    _page_size: ClassVar[int] = 12
```
Comment (matin) [inline src/garth/stats/training_status.py:158]: break up resources into different files similar to body battery

File: src/garth/stats/training_status.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                for entry in device_data:
+                    if isinstance(entry, dict):
+                        # Flatten the acuteTrainingLoadDTO data
+                        acute_load = entry.pop("acuteTrainingLoadDTO", {})
+                        if isinstance(acute_load, dict):
+                            entry.update(acute_load)
+                        result.append(entry)
+                return result
+
+        return []
```
Comment (matin) [inline src/garth/stats/training_status.py:94]: the final return statement should return the result of the happy path to make it clearer to read the code

**Review by matin — COMMENTED** [review]
@claude mostly there, but there are some changes I commented on

**Comment by matin** [comment]
@coderabbitai review

**Comment by matin** [comment]
> @matin does this fetch the training plans suggested by Garmin Coach? If not do you know how to fetch those? (Suggested Training)

I don't believe so. I've been behind on Garth, but I'm going to be catching up this week. I'll look into Garmin Coach as well
